### DISCUSSION TITLE
telegram and a MQTT TLS

### DISCUSSION THREAD

**Discussion by biotecov** [arendst/Tasmota#17697] [Q&A]
Hello, is possible to use simultaneously Telegram and MQTT TLS? Telegram is mostly undocumented, so I cannot find any clue on the docs.

**Comment by barbudor** [comment]
As you found out, you need to enabled Fingerprint mode (SetOption132 ) to use Telegram
So as [documented](https://tasmota.github.io/docs/TLS/), Tasmota can support 2 different fingerprints.
I would assume, without having tested it myself, that might allow to have 1 fingerprint for MQTT and 1 for Telegram.

**Reply by biotecov** [reply]
The important part is that I can confirm that it works. I was able to compile a firmware with TLS and telegram support. Now I have both protocols enabled and working, and I could connect to an external TLS broker and to telegram. The gray area is that I don't know how I did and what are the exact steps for sure :) SetOption132 enables fingerprint ... but which one? both? I know the telegram fingerprint is hard coded, but it seems that so132 1 is required to make use of it and have telegram connected, whereas the mqttfingerprint1 must start as 0 and is learned when so132 is set. I am confused about the utility of mqttfingerprint2 and its preset values ... anyway, I don't have the time to investigate these details and I can live with these doubts.
